src/index.ts

```diff
@@ -424,9 +498,9 @@ class ThreadPool {
     filename = maybeFileURLToPath(filename);
 
     let signal: AbortSignalAny | null;
-    if (this.closingUp) {
+    if (this.closingUp || this.destroying) {
```

Comment:
What is the difference between closingUp and destroying flag? I think the two can be merged.